src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -512,6 +512,25 @@ bool usesOriginalInput(BoundDecisionDagNode node)
 
                 return rewrittenDag;
 
+                static BoundExpression getExpression(int index, BoundObjectCreationExpression expression)
```

Comment:
nit: Consider folding this logic into the above loop, so that this gets repeated fewer times. This is an uncommon scenario, so readability should come first (try and see what it looks like).

Reply:
@jcouv Based on https://github.com/dotnet/roslyn/pull/47893#issuecomment-696386895, this method may be needed in another place. In such a case, it would make more sense to be extracted to an extension method somewhere.